client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -14,23 +14,63 @@ import { connect } from 'react-redux';
 import Card from 'components/card';
 import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
 import FormFieldset from 'components/forms/form-fieldset';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
 import isJetpackModuleUnavailableInDevelopmentMode from 'state/selectors/is-jetpack-module-unavailable-in-development-mode';
 import isJetpackSiteInDevelopmentMode from 'state/selectors/is-jetpack-site-in-development-mode';
+import { activateModule, deactivateModule } from 'state/jetpack/modules/actions';
+import isJetpackModuleActive from 'state/selectors/is-jetpack-module-active';
+import isActivatingJetpackModule from 'state/selectors/is-activating-jetpack-module';
+import isDeactivatingJetpackModule from 'state/selectors/is-deactivating-jetpack-module';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getSiteSlug } from 'state/sites/selectors';
 import SupportInfo from 'components/support-info';
 
 class SpeedUpSiteSettings extends Component {
+	static defaultProps = {
+		togglingSiteAccelerator: false,
+	};
+
 	static propTypes = {
 		fields: PropTypes.object,
 		isRequestingSettings: PropTypes.bool,
 		isSavingSettings: PropTypes.bool,
 		jetpackVersionSupportsLazyImages: PropTypes.bool,
+		siteAcceleratorSupported: PropTypes.bool,
+		activateModule: PropTypes.func,
+		deactivateModule: PropTypes.func,
 
 		// Connected props
 		photonModuleUnavailable: PropTypes.bool,
 		selectedSiteId: PropTypes.number,
 		siteSlug: PropTypes.string,
+		photonModuleActive: PropTypes.bool,
+		assetCdnModuleActive: PropTypes.bool,
+		togglingSiteAccelerator: PropTypes.bool,
+	};
+
+	handleCdnChange = () => {
+		const { photonModuleActive, assetCdnModuleActive, selectedSiteId } = this.props;
+
+		// Check if any of the CDN options are on.
+		const cdnStatus = photonModuleActive || assetCdnModuleActive;
+
+		// If one of them is on, we turn everything off.
+		if ( true === cdnStatus ) {
+			if ( false === ! photonModuleActive ) {
+				this.props.deactivateModule( selectedSiteId, 'photon' );
+			}
+			if ( false === ! assetCdnModuleActive ) {
+				this.props.deactivateModule( selectedSiteId, 'photon-cdn' );
```

Comment:
I'm not sure what's happening, when deactivating Site Accelerator, I keep getting an error for one of the deactivation requests:

![](https://cldup.com/mnHfQ2pJOR.png)

Inspecting the network request reveals that the module was already inactive:

![](https://cldup.com/AJRM8W1k9Q.png)

This is odd, because I just had the module activated, as I previously activated the "Site Accelerator", and it activated both modules.

Reply:
I got this to happen as well at times, but it does not seem to be consistent (or I could not pinpoint the exact combination of factors to make that happen. I don't understand how it happens yet.

Reply:
For me it fails like that. For a few minutes I thought I had the exact set of steps to reproduce, but it wasn't the case.

Reply:
ed7ccddf89218f8c03dcf9d1e8543a7db1a45228 appears to be a better approach, I can't seem to get that error anymore.

Reply:
I second that - couldn't repro the error anymore. Nice!